### PR TITLE
[catmem] Performance counters

### LINKED ISSUE #894: [catmem] Add performance counters
## Context

Catmem needs profile and timer statements similar to Catnap.

### PR DESCRIPTION
**Pull request by carvalhof**
This PR closes #894.

Commits:

- 323979bd4310fbc0ae53904579e7f8a433ad5c47 - Inserting the profiler annotations.
- c6943e6b94fab5705af7688d840b96eb369a051a - Updating the pipe-push-pop example.
- 155259014b515ff4f0f0f98e697705a8c448a84c - Updating the pipe-ping-pong example.

### CODE REVIEW

**Review by iyzhang — APPROVED** [review]
Thanks for adding this @carvalhof! Should we add this profiler printing to all of our tests? Then we could gather them and display a performance dashboard.

File: src/rust/catmem/mod.rs [hunk 1]
```diff
@@ -131,6 +146,8 @@ impl CatmemLibOS {
         qd: QDesc,
         yielder: Yielder,
     ) -> (QDesc, OperationResult) {
+        #[cfg(feature = "profiler")]
+        timer!("catmem::close_coroutine");
```
Comment (ppenna) [inline src/rust/catmem/mod.rs:150]: No need to benchmark the async co-routine as well, as it may take time to finish up.

**Comment by ppenna** [comment]
@carvalhof would you remove the profiling from asynchronous code? Those we want to measure from application side. @iyzhang please jump in if you disagree.

File: src/rust/catmem/mod.rs (hunk 1 above)
Comment (carvalhof) [reply src/rust/catmem/mod.rs]: Removed in 6c6f6b77268c73907501add92ea403fcea4eb0e6.

**Comment by carvalhof** [comment]
> @carvalhof would you remove the profiling from asynchronous code? Those we want to measure from application side. @iyzhang please jump in if you disagree.

Done in 6c6f6b77268c73907501add92ea403fcea4eb0e6.

**Review by ppenna — APPROVED** [review]

**Pull request merged**